doc/user_guide/api.rst

```diff
@@ -166,6 +166,7 @@ API Functions
    topo_feature
    value
    vconcat
+   when
```

Comment:
I expected `When` would also show up after 58e005406e42268ec0ffd63c38d077384685375a but there are 2 issues. 

1. It would never be selected under the current conditions of `generate_api_docs` https://github.com/vega/altair/blob/76a9ce1e27a21359e7b0ef9e6cb865f8d95fc298/tools/generate_api_docs.py#L122-L133
2. If this was altered to include classes from `api`, there would be a collision with `when`/`When` on the url.

I'm still fairly new to `sphinx` so there may be a simple solution to the latter. Due to how `api` is defined, moving these classes to another module would require some less than desirable import special casing.

Reply:
See https://github.com/vega/altair/pull/3427#discussion_r1659965947